web-client/public/js/api/grnsight-api.js

```diff
@@ -19,6 +19,36 @@ const buildExpressionURL = function (selection, genes) {
     baseQuery;
 };
 
+
+const buildNetworkGenesQuery = (genes) => {
+    let result = "";
+    if (Array.isArray(genes)) {
+        for (let gene of genes) {
+            result += `${gene},`;
+        }
+    } else {
+        for (let gene in genes) {
+            result += `${gene},`;
+        }
+    }
+    return result.substring(0, result.length - 1);
+};
+
+
+const buildAltExpressionURL = function (queryType, queryInfo) {
+    let baseQuery = `expressiondb?type=${queryType}`;
+    if (queryInfo !== null) {
+        for (let header in queryInfo) {
+            if (header === "genes") {
+                baseQuery += `&${header}=${buildNetworkGenesQuery(queryInfo[header])}`;
+            } else {
+                baseQuery += `&${header}=${queryInfo[header]}`;
+            }
+        }
+    }
```

Comment:
I think there is a simpler way to build this—see `URLSearchParams` https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
So, I tried implementing this change, but I keep on getting 404 not found errors. I printed the new url string and it looked fine, but the actual network call is returning errors.
![Screenshot from 2022-11-09 10-56-03](https://user-images.githubusercontent.com/21343072/200916981-7e05b4d2-3a88-4296-9732-7df8fa00d9e8.png)